dandi/download.py

```diff
@@ -757,22 +795,24 @@ def __exit__(self, exc_type, exc_value, traceback):
                     rmtree(self.filepath)
                     self.writefile.replace(self.filepath)
         finally:
+            assert self.lock is not None
             self.lock.release()
             if exc_type is None:
                 rmtree(self.dirpath, ignore_errors=True)
             self.lock = None
             self.fp = None
             self.offset = None
-        return False
 
-    def append(self, blob):
+    def append(self, blob: bytes) -> None:
+        if self.fp is None:
+            raise ValueError("DownloadDirectory.append() called outside of context")
```

Comment:
```suggestion
            raise ValueError("DownloadDirectory.append() called outside of context manager")
```
just to make it clearer